parse.go

```diff
@@ -467,8 +505,12 @@ func (p *parser) parseFileHeader() []*element.Element {
 
 	// check for magic word
 	if s := p.decoder.ReadString(4); s != "DICM" {
-		p.decoder.SetError(errors.New("Keyword 'DICM' not found in the header"))
-		return nil
+		if p.force {
+			dicomlog.Vprintf(1, "WARNING Keyword 'DICM' not found in the header. Force enabled, therefore continuing...")
```

Comment:
I think you're right. Looking at pydicom in [read_preabmle](https://github.com/pydicom/pydicom/blob/master/pydicom/filereader.py#L571), which has a force option, it appears that if no header is found (as distinguished by the missing 'DICM' tag, the program [rewinds](https://github.com/pydicom/pydicom/blob/master/pydicom/filereader.py#L619) back to byte 0 before continuing so that it's not 128 bytes ahead. I think it's assumed in that code that if the DICM keyword is missing then the entire header is missing and the file starts with actual data.

I think a good way to approach this, as you said, might be to have a more specific flag that makes the assumption that the header data is missing and treats the first 128 + 4 bytes as tags. I think that would probably be the easiest way to implement on this side, it wouldn't necessarily have to break backwards compatibility, and it would be easy to implement on the people using this library's side. Since we already handle error checking, it could be something as simple as:

p, err := dicom.NewParserFromFile(...)
if err != nil {
       p, err := dicom.NewParserFromFile(skipPreamble = true) {
              if err != nil { return}
}
// keep using parser here with first 128 + 4 bytes treated as tags in Parse

@wkoszek knows more about what these files look like than I do, but if these dicoms are not completely missing the header and we treat those bytes as tags, will this cause issues in Parse() or elsewhere?

Pydicom also has the force option applicable for the MetaElementGroupLength tag missing ([issue 53](https://github.com/suyashkumar/dicom/issues/53)), which I haven't had a chance to take a look at yet but might be a useful reference. I think it has something to do with [this](https://github.com/pydicom/pydicom/blob/master/pydicom/filereader.py#L212) code chunk, but I haven't looked too closely

Reply:
@suyashkumar Provided a sample dump on a ticket.

Reply:
Thanks @wkoszek! @IChocked: Let's proceed with my earlier recommendation that if NewParserForce is called, we skip reading the preamble entirely (Maybe we should call it NewParserSkipPreamble). 

Probably the cleanest way to do this is just don't call `metaElems := p.parseFileHeader()` at all in the Force version. 

Under the hood, remember to have NewParserForce and NewParser call newParserInternal() which would take an optional skipPreamble parameter. We may have to call p.decoder.PushTransferSyntax(binary.LittleEndian, dicomio.ExplicitVR) to set the default transfer syntax in this scenario.